### DISCUSSION TITLE
Memory Issues with FabricJS PDF Annotation

### DISCUSSION THREAD

**Discussion by MallauryGerard** [fabricjs/fabric.js#9555] [Q&A]
Hello,

I'm using FabricJS to annotate PDF files online. I generate the PDF file as I scroll through it (2 pages at a time). Above each page, I add a canvas for adding text.

I'm encountering two memory-related problems:
- To improve the PDF quality, I load it into a 2000px wide canvas (which I later reduce to 1000px using CSS).
- The more I scroll (loading more pages), the more the browser freezes. Interestingly, this freezing doesn't occur when I generate pages with a 100px canvas, for example.

I'm using the dev console to analyze memory usage and compare the 2000px canvas with the 100px canvas.

**My question is: Why is there no difference in memory usage between a 100px and a 2000px canvas, even though the browser freezes with the larger canvas?**
And Should I consider deleting the canvas from previous pages as I scroll to free up memory?

For instance, when I am loading a 24MB PDF :
- Heap memory is : 75MB (2 pages loaded)
- Heap memory is : 145MB (8 pages loaded)
These values are the same with 2000px or 100px canvas...

Thanks!

**Comment by ShaMan123** [comment]
It is rather impossible to say without a repro and a perf flame graph.
I would check the pdf lib first because it is a heavy operation and known to block the main thread. Consider doing it offscreen in a web worker context.
